sdk/core/azure-core/src/main/java/com/azure/core/http/rest/PagedFlux.java

```diff
@@ -63,4 +66,27 @@ public PagedFlux(Supplier<Mono<PagedResponse<T>>> firstPageRetriever,
                      Function<String, Mono<PagedResponse<T>>> nextPageRetriever) {
         super(firstPageRetriever, nextPageRetriever);
     }
+
+    /**
+     * Maps this PagedFlux instance of T to a PagedFlux instance of type S as per the provided mapper function.
+     *
+     * @param mapper The mapper function to convert from type T to type S.
+     * @param <S> The mapped type.
+     * @return A PagedFlux of type S.
+     */
+    public <S> PagedFlux<S> mapPage(Function<T, S> mapper) {
+        return new PagedFlux<S>(() -> getFirstPageRetriever().get()
+            .map(mapPagedResponse(mapper)),
+            continuationToken -> getNextPageRetriever().apply(continuationToken)
+                .map(mapPagedResponse(mapper)));
+    }
+
+    private <S> Function<PagedResponse<T>, PagedResponse<S>> mapPagedResponse(Function<T, S> mapper) {
+        return pagedResponse -> new PagedResponseBase<HttpRequest, S>(pagedResponse.getRequest(),
+            pagedResponse.getStatusCode(),
+            pagedResponse.getHeaders(),
+            pagedResponse.getValue().stream().map(mapper).collect(Collectors.toList()),
+            pagedResponse.getContinuationToken(),
+            null);
```

Comment:
What does this `null` correspond to?

Reply:
The parameter here is for a second generic type on the PagedResponse class for deserialized headers which is a property mapped class of headers.

@JonathanGiles I've pondering why we have this as part of the PagedResponse interface/abstract class, I've only seen them in autogen SDKs.

Reply:
So `PagedResponseBase`, which Srikanta is creating here, is not public API. It implements the `PagedResponse` API, which is what users should only ever see. This second parameter type is only on PagedResponseBase, and is not visible to end users. There is probably scope to clean this up, but it isn't user-facing, so it is low priority in my eyes.